packages/client/src/tlog/verify/checkpoint.ts

```diff
@@ -196,36 +188,38 @@ class LogCheckpoint {
 function filterTLogInstances(
   tlogInstances: sigstore.TransparencyLogInstance[],
   integratedTime: string
-): ValidTransparencyLogInstance[] {
+): sigstore.ViableTransparencyLogInstance[] {
   const targetDate = new Date(Number(integratedTime) * 1000);
-  return tlogInstances.filter((tlog): tlog is ValidTransparencyLogInstance => {
-    // Must have a log ID
-    if (!tlog.logId) {
-      return false;
-    }
+  return tlogInstances.filter(
+    (tlog): tlog is sigstore.ViableTransparencyLogInstance => {
```

Comment:
This is a neat trick . . . specifying a type assertion in the return value of the anonymous function called by `Array.filter` allows us to narrow the type of the return value.  We call `filter` on an array of `TransparencyLogInstance`s but the returned array contains `ViableTransparencyLogInstance`s (i.e. instances where we've validated that the required log ID and public key are present)